### ISSUE TITLE
cinterop: bool* not supported as parameter

### ISSUE THREAD

**Issue by nlbuescher** [JetBrains/kotlin-native#2675]
Using Kotlin 1.3.21 MPP

If I have a C or C++ function
```c
// test.h
#include <stdbool.h>
void test(bool* pb)
{
    *pb = true;
}
```

the corresponding Kotlin stub generated by cinterop comes out as
```kotlin
// ctest.kt
@Deprecated("Unable to import this declaration", level = DeprecationLevel.ERROR)
fun test(pb: CValuesRef<BooleanVar>?): Unit = throw UnsupportedOperationException()
```

I am able to work around the issue by defining a wrapper function in the `.def`
```c
// ctest.def
void _test(int* pi)
{
    test((bool*)pi);
}
```

As addressed in #756 and #794, C bool is supported (as also evidenced by the fact that `bool*` is correctly mapped to `CValuesRef<BooleanVar>?`) but for some reason the stub generator is having problems with this and no `kniBridge` function is generated.

EDIT: added `#include <stdbool.h>` to `test.h` because it wouldn't compile otherwise if you were to try it out.

**Comment by olonho** [comment]
C interop works with C headers, there's not type `bool` in C, unless `<stdbool.h>` include is used. But indeed there's a subtle problem related to translation rules of boolean types, somewhat straightened up in https://github.com/JetBrains/kotlin-native/pull/2676

**Comment by msink** [comment]
https://github.com/JetBrains/kotlin-native/blob/master/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CSyntaxSupport.kt#L29

**Comment by nlbuescher** [comment]
@olonho the issue seems to be specific to `bool*`, since a function with a `bool` parameter is translated properly and can be used, but any function with a `bool*` parameter is marked as Deprecated, even with `#include <stdbool.h>` (double checked just to be sure that wasn't the problem)

**Comment by olonho** [comment]
Shall be fixed with #2676, please check.
